Sage/carbon
# fix: remove border style when the input is disabled

### Proposed behaviour

This PR tries to solve the issue [4631](https://github.com/Sage/carbon/issues/4631). The behaviour proposed when an `Input` component gets disabled then the outline is removed and the focus is passed to the following element in the DOM. 
 

### Current behaviour

When an `Input` component (or components that contains an `Input`) are disabled then the focus remains in the disabled element, which is the `Input` and once `tab` is pressed the focus is removed however the outline sticks to the element. That create two elements with an outline.

![image](https://user-images.githubusercontent.com/11317965/146033555-914d9138-d122-4d41-87c7-bb8b1dfc31bd.png)


### Checklist


#### QA


### Additional context

None

### Testing instructions

1. Make sure the outline is removed from the `Input` when the component is disabled.
2. Add different elements surrounding an `Input` and once it's disabled, the next focusable element should be focused.

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/affectionate-pine-vcox0?file=/src/index.js
